postgres_stats.c

```diff
@@ -1112,6 +1124,16 @@ static void get_pg_stat_activity(pg_stat_activity_list *pg_stats)
 
 	MemoryContextDelete(othercxt);
 
+	pg_stats->wal_metrics.is_wal_replay_paused = RecoveryIsPaused();
+	pg_stats->wal_metrics.last_xact_replay_timestamp = GetLatestXTime();
+
+	pg_stats->wal_metrics.last_wal_replay_lsn = GetXLogReplayRecPtr(NULL);
+	pg_stats->wal_metrics.current_wal_lsn = GetXLogWriteRecPtr();
+	#if PG_VERSION_NUM >= 130000 
+		pg_stats->wal_metrics.last_wal_receive_lsn = GetWalRcvFlushRecPtr(NULL, NULL);
+	#else
+		pg_stats->wal_metrics.last_wal_receive_lsn = GetWalRcvWriteRecPtr(NULL, NULL);
+	#endif
```

Comment:
```suggestion
	if (pg_stats.recovery_in_progress) {
		pg_stats->wal_metrics.is_wal_replay_paused = RecoveryIsPaused();
		pg_stats->wal_metrics.last_xact_replay_timestamp = GetLatestXTime();
		pg_stats->wal_metrics.last_wal_replay_lsn = GetXLogReplayRecPtr(NULL);
#if PG_VERSION_NUM >= 130000 
		pg_stats->wal_metrics.last_wal_receive_lsn = GetWalRcvFlushRecPtr(NULL, NULL);
#else
		pg_stats->wal_metrics.last_wal_receive_lsn = GetWalRcvWriteRecPtr(NULL, NULL);
#endif
	} else pg_stats->wal_metrics.current_wal_lsn = GetXLogWriteRecPtr();
```

And logically this block should be in the get_postgres_stats() function.